### PR TITLE
Dynamic and improved aggregated lists (Portfolio)

### PR DESCRIPTION
**Pull request by Remi749**
### Your checklist for this pull request

- [X] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your main!
- [X] Make sure you are making a pull request against the **dev** branch (left side). Also you should start *your branch* off *dev*.
- [X] Check the commit's or even all commits' message 
- [X] Check if your code additions will fail linting checks
- [X] Remember: Add issue description to [CHANGELOG](https://github.com/Puzzlepart/prosjektportalen365/blob/dev/CHANGELOG.md) with the **ID of the issue** associated with this PR
- [X] Documentation: Have a look at the [PP365 User manual](https://puzzlepart.github.io/prosjektportalen-manual/) and consider the need for updates to be made. Updates can be done directly into the 'Kladd' branch or by providing information to test team for implementation.

### Description

A big change to the way the aggregated webparts at the Portfolio level works. Implementation of the same functionality found in the PortfolioOverview webpart. This includes Views, filtering, grouping, etc...

Aggregated webparts included in Prosjektportalen 365 that will be affected:
- Gevinstoversikt
- Erfaringslogg
- Leveranseoversikt
- Risikooversikt

**NOTE!**
The improvements made here does not affect Program/Overordnet. The aggregated webparts that came with 'Program- and Overordnet-område' are created as unique webparts instead of using the Portfolio aggregated webpart directly and use the 'Datakilder' list to set up the source.

Because of this we will have to keep Program and Project at the pp365-portfoliowebpart@1.5.4-679.2 version until we have created new webparts for Program/Overordnet that will work directly and more seamlessly with the new additions made here.

## Additions:
- Added button to 'Konfigurasjon av Prosjektportalen' page.
- Added a new list.

## Provision template additions/changes:
- New List (*Title, Url added to resource files*):
  - Lists/Prosjektinnholdskolonner

- New SiteFields (*Name, Descriptions added to resource files.*)
  - GtODataQuery (Note field to filter the projects found within the datasource searchquery)
  - GtProjectContentColumns (Lookup to 'Prosjektinnholdskolonner')
  - GtProjectContentGroupBy (Lookup to 'Prosjektinnholdskolonner')
  - GtProjectContentRefiners (Lookup to 'Prosjektinnholdskolonner')

- Changes to Datakilder
  - Added the new SiteFields to the list
  - Reorder of columns in list

- Changes to datakilde contenttype
  - Added the new SiteFields to the contenttype

- Changes to the following ClientSidePages
  - Erfaringslogg, Leveranseoversikt, Risikooversikt, Gevinstoversikt
    - Removed all data for columns property in the webpart
    - Set Overwrite to true
  - Gevinstoversikt
    - Switched out the webpart to use aggregated webpart instead of the benefits webpart
  - Konfigurasjon
    - Added button (QuickLink) for 'Prosjektinnholdskolonner' list to 'Konfigurasjon av Prosjektportalen' page.

## Changes

**Aggregated webpart**

- Views
  - Ability to create new views
  - Possibility to see the views from 'Datakilder' in the drop-down menu with 'Edit' functionality.
image
  
- Filtering
  - Ability to filter the view based on the selected columns from 'Datakilder'
image

- Group by
  - Ability to set a pre-defined group by column based on the selected column from 'Datakilder'

- Change in 'Datakilder' list (4 new columns)
image

- Columns
  - Columns defined in 'Datakilder' will be used instead of the columns found in webpart properties
  - Columns are editable same as before where all data is persisted

- Edit mode
    - 'Legg til kolonne':
      - This will now show a dropdown instead of opening the Column panel (Legg til kolonne/Vis/skjul kolonner)
      - When creating a new column, the column is also created in the 'Prosjektinnholdskolonner' list and connected to the 'Datakilde' element for the current view
      - Deleting a column will also remove this from the 'prosjektinnholdskolonner' list
        - Columns: 'Prosjekt' and 'Tittel' can't be deleted.
      - Editing the name for the columns in an aggregated list will only persist the changes directly in the list and won't affect the element in the 'Prosjektinnholdskolonner' list
    - Added new option: 'Vis/skjul kolonner', ability to remove columns from Datakilder for the current view
      - Tried to replicate this to be identical to SharePoint out of the box version

### How to test

Test out the changes made the aggregated webpart

1. Check to see if 'Prosjektinnholdskolonner' and 'Datakilder' is correctly set up after upgrade
2. Test creating, changing and editing views
3. Test filtering
4. Test group by
5. Edit columns and test deleting custom columns
6. Test 'vis/skjul kolonner'
7. Make changes to 'Datakilder' list
   a. Test odata query

### Relevant issues (if applicable)

#706

### CODE REVIEW

**Comment by Remi749** [comment]
@olemp @tarjeieo, when do you think you have time to review or merge code?

**Review by tarjeieo — APPROVED** [review]
In god we trust

**Comment by olemp** [comment]
> @olemp @tarjeieo, when do you think you have time to review or merge code?

I'll have a look on monday.

File: SharePointFramework/PortfolioWebParts/package.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "object-assign": "4.1.1",
     "office-ui-fabric-react": "6.214.0",
     "pp365-projectwebparts": "latest",
-    "pp365-shared": "latest",
+    "pp365-shared": "1.5.4-706.10",
```
[the remark this answers is not in the document]
Comment (Remi749) [reply SharePointFramework/PortfolioWebParts/package.json:42]: PortfolioWebParts skal bruke latest ja

**Review by olemp — APPROVED** [review]
Looks good! Let's get this into dev.

**Pull request merged**